llvm/lib/Analysis/LazyValueInfo.cpp

```diff
@@ -641,7 +641,7 @@ LazyValueInfoImpl::solveBlockValueImpl(Value *Val, BasicBlock *BB) {
   // instruction is placed, even if it could legally be hoisted much higher.
   // That is unfortunate.
   PointerType *PT = dyn_cast<PointerType>(BBI->getType());
-  if (PT && isKnownNonZero(BBI, DL))
+  if (PT && isKnownNonZero(BBI, /*Depth=*/0, SimplifyQuery(DL)))
```

Comment:
FWIW, you could just pass `DL` here and other places that use only DL. It will be converted implicitly. But this is fine.